modules/payloads/singles/windows/x64/messagebox.rb

```diff
@@ -0,0 +1,198 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+module MetasploitModule
+
+  CachedSize = 299
+
+  include Msf::Payload::Windows
+  include Msf::Payload::Single
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'          => 'Windows MessageBox x64',
+      'Description'   => 'Spawn a dialog via MessageBox using a customizable title, text & icon',
+      'Author'        => [
+        'pasta <jaguinaga[at]infobytesec.com>'
+      ],
+      'License'       => GPL_LICENSE,
+      'Platform'      => 'win',
+      'Arch'          => ARCH_X64,
+    ))
+    register_options([
+      OptString.new('TITLE', [ true, "Messagebox Title", "MessageBox" ]),
+      OptString.new('TEXT', [ true, "Messagebox Text", "Hello, from MSF!" ]),
+      OptString.new('ICON', [ true, "Icon type can be NO, ERROR, INFORMATION, WARNING or QUESTION", "NO" ]),
```

Comment:
Check out OptEnum

Reply:
the new version was pushed with the icon type filter (using OptEnum) and also solved almost everything the rubocop's issues

this is the response of rubocop until now:

```Inspecting 1 file
C

Offenses:

modules/payloads/singles/windows/x64/messagebox.rb:6:1: C: Metrics/ModuleLength: Module has too many lines. [179/100]
module MetasploitModule ...
^^^^^^^^^^^^^^^^^^^^^^^
modules/payloads/singles/windows/x64/messagebox.rb:7:1: C: Layout/EmptyLinesAroundModuleBody: Extra empty line detected at module body beginning.
modules/payloads/singles/windows/x64/messagebox.rb:8:3: C: Naming/ConstantName: Use SCREAMING_SNAKE_CASE for constants.
  CachedSize = 299
  ^^^^^^^^^^
modules/payloads/singles/windows/x64/messagebox.rb:15:9: C: Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
        'Name'          => 'Windows MessageBox x64',
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
modules/payloads/singles/windows/x64/messagebox.rb:15:9: C: Layout/AlignParameters: Use one level of indentation for parameters following the first line of a multi-line method call.
        'Name'          => 'Windows MessageBox x64', ...
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
modules/payloads/singles/windows/x64/messagebox.rb:16:9: C: Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
        'Description'   => 'Spawn a dialog via MessageBox using a customizable title, text & icon',
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
modules/payloads/singles/windows/x64/messagebox.rb:17:9: C: Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
        'Author'        => [ ...
        ^^^^^^^^^^^^^^^^^^^^
modules/payloads/singles/windows/x64/messagebox.rb:20:9: C: Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
        'License'       => GPL_LICENSE,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
modules/payloads/singles/windows/x64/messagebox.rb:21:9: C: Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
        'Platform'      => 'win',
        ^^^^^^^^^^^^^^^^^^^^^^^^
modules/payloads/singles/windows/x64/messagebox.rb:22:9: C: Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
        'Arch'          => ARCH_X64)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 10 offenses detected
```
I feel that this corrections are againts the guidelines, what do you think? feel free to paste a snippet

Reply:
Those warnings are for the module metadata boilderplate only and can safely be ignored. Most, if not all, modules generate the same warnings.

Although, the keys in the `super` call (`Name`, `Description`, etc) are usually indented only two spaced, rather than four.

Reply:
thank @bcoles, I cleaned the spaces that survived to the purge in [c8740a1](https://github.com/rapid7/metasploit-framework/pull/10834/commits/c8740a1b0f3aec337ef7fe3e19c4d12047a8787b) commit